applications/ParticleMechanicsApplication/tests/restart_tests.py

```diff
@@ -0,0 +1,121 @@
+            self.start_time = self.project_parameters_save["problem_data"]["start_time"].GetDouble()
+            self.end_time = self.project_parameters_save["problem_data"]["end_time"].GetDouble()
+
+            self.project_parameters_save["solver_settings"]["time_stepping"]["time_step"].SetDouble(self.time_step)
+            self.project_parameters_save["problem_data"]["start_time"].SetDouble(self.start_time)
+            self.project_parameters_save["problem_data"]["end_time"].SetDouble(self.end_time)
+
+            # Now clone the settings after the common settings are set
+            self.project_parameters_load = self.project_parameters_save.Clone()
+
+            # Adding the specific settings (minimal to test the default settings)
+            save_restart_parameters = KratosMultiphysics.Parameters("""{
+                "restart_processes" : [
+                    {
+                    "python_module"   : "save_restart_process",
+                    "kratos_module"   : "KratosMultiphysics",
+                    "process_name"    : "SaveRestartProcess",
+                    "Parameters"            : {
+                    }
+                }]
+             }""")
+
+            model_part_name = self.project_parameters_save["solver_settings"]["model_part_name"]
+            save_restart_parameters["restart_processes"][0]["Parameters"].AddValue("model_part_name", model_part_name)
+
+            self.project_parameters_save.AddValue("output_processes", save_restart_parameters)
+
+            load_mp_import_settings = self.project_parameters_load["solver_settings"]["model_import_settings"]
+            load_mp_import_settings.AddEmptyValue("restart_load_file_label")
+            load_mp_import_settings["input_type"].SetString("rest")
+            load_mp_import_settings["restart_load_file_label"].SetString(str(self.start_time+self.time_step))
+
+            # Correct the path
+            load_mp_import_settings["input_filename"].SetString("MPM_Material")
+
+    def test_execution(self):
+        # Within this location context:
+        with KratosUnittest.WorkFolderScope("", __file__):
+            model_save = KratosMultiphysics.Model()
+            model_load = KratosMultiphysics.Model()
+            particle_mechanics_analysis.ParticleMechanicsAnalysis(model_save, self.project_parameters_save).Run()
+            particle_mechanics_analysis.ParticleMechanicsAnalysis(model_load, self.project_parameters_load).Run()
+
+    def tearDown(self):
+        # Within this location context:
+        with KratosUnittest.WorkFolderScope("", __file__):
+            # remove the created restart files
+            #raw_path, raw_file_name = os.path.split(self.file_name)
+            raw_file_name = "MPM_Material"
+            folder_name = raw_file_name + "__restart_files"
+
+            self.addCleanup(kratos_utils.DeleteDirectoryIfExisting, GetFilePath(folder_name))
```

Comment:
This needs to be added before running the tests, otherwise it won't work 

(The example I referenced also is not correct)

Reply:
Ok I see :sweat_smile: 
So I should add this just before the tests are run in the test_execution function?

Reply:
Like in f3cac64

Reply:
@philbucher could you please confirm and review again when you have time so that we can merge the PR. Thanks! 😊

Reply:
Yep now it's correct 
Sorry for the delay